presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -541,6 +543,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            Optional<Expression> argumentNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
Thank you for the clarification. I'm fine with making it strict for now. I added a test in `TestAnalyzer` that fails because `grouping()` is referencing an output column but I kept a test in `AbstractTestQueries` where `grouping()` refers to a base column.